### ISSUE TITLE
Extension stops working after some time

### ISSUE THREAD

**Issue by eduardosm** [kazysmaster/gnome-shell-extension-disable-unredirect#17]
After gnome has been running for some time, the extension stops working, requiring to disable and enable it again to make it work again.

**Comment by aunetx** [comment]
I forked this extension to prevent this bug, you can find it [here](https://github.com/aunetx/gnome-shell-extension-disable-unredirect).

It should be soon reviewed for extensions.gnome.org.

**Comment by kazysmaster** [comment]
hi @aunetx, have you considered making a pull request?

**Comment by aunetx** [comment]
Hi, sorry I did not even considered that, don't know why I did not think about it!

If you're OK to include these changes to your extension, I will make a PR with these and remove my extension (don't know if it is possible to remove from extensions.gnome.org, this would be stupid to duplicate the same extension)

**Comment by kazysmaster** [comment]
Nah, it's fine, let it be. I'm not using this extension myself now so I'm not updating it. It's nice to see when someone steps up and works on it. There are other forks so I thought others might notice changes in this repo and benefit from them.

**Comment by Movisivi** [comment]
I read that this issue was solved in kernel 3.38. Does this mean that I don't need this as of Fedora 34?

**Comment by ryanmusante** [comment]
Is this still active?

**Comment by Movisivi** [comment]
> Is this still active?

This is not  needed if you are on Wayland.

**Comment by ryanmusante** [comment]
I'm using Nvidia + Gnome so stuck on Xorg in the meantime. Does this build include the timeout fix?

**Comment by kazysmaster** [comment]
btw, according to the source https://gitlab.gnome.org/GNOME/mutter/-/blob/main/src/compositor/compositor.c#L1322 there is a counter of invocations, so there's maybe something else that enables the redirection after login.

**Comment by Stez69** [comment]
Hey I'm on Wayland and am experiencing this bug on Gnome 45 does anyone have a fix

**Comment by Karamveer9200** [comment]
@Stez69 check my pull request, I was facing the same issue in gnome 45, it seems to fixed after some tweaking.

**Comment by kazysmaster** [comment]
I've merged the PR, I think it is a good approach., pls test. I will submit it at some point tomorrow to extensions.gnome.org

**Comment by kazysmaster** [comment]
Changes didn't go through a code review when submitting to extensions.gnome.org. The reviewer did suggest a monkey patching - which I did in https://github.com/kazysmaster/gnome-shell-extension-disable-unredirect/tree/monkey_patch 
please test that branch if anyone can reproduce the problem.
